stan/math/prim/fun/wiener5_lpdf.hpp

```diff
@@ -0,0 +1,561 @@
+    lg1 = (eta_sqr * pow(a * w, 2) - 2 * a * v * w - pow(v, 2) * q) / 2.0 / temp
+          - 2 * log(a) - 0.5 * log(temp);
+  } else {
+    lg1 = (-2 * a * v * w - pow(v, 2) * q) / 2.0 - 2 * log(a);
+  }
+  T_return_type es = (err - lg1);
+  T_return_type K1 = (sqrt(2.0 * q_asq) + w) / 2.0;
+  T_return_type u_eps
+      = fmin(-1.0, LOG_TWO + LOG_PI + 2.0 * log(q_asq) + 2.0 * (es));
+  T_return_type arg = -q_asq * (u_eps - sqrt(-2.0 * u_eps - 2.0));
+  T_return_type K2 = (arg > 0) ? 0.5 * (sqrt(arg) - w) : K1;
+  kss = ceil(fmax(K1, K2));
+
+  // calculate the number of terms needed for large t
+  T_return_type el = es;
+  K1 = 1.0 / (pi() * sqrt(q_asq));
+  K2 = 0.0;
+  T_return_type temp_ = -2.0 * (log(pi() * q_asq) + el);
+  if (temp_ >= 0)
+    K2 = sqrt(temp_ / (PISQ * q_asq));
+  kll = ceil(fmax(K1, K2));
+
+  // if small t is better
+  if (2 * kss <= kll) {
+    T_return_type fplus = NEGATIVE_INFTY;
+    T_return_type fminus = NEGATIVE_INFTY;
+    T_return_type twot = 2.0 * q_asq;
+    if (static_cast<size_t>(kss) > 0) {
+      for (size_t k = static_cast<size_t>(kss); k >= 1; k--) {
+        T_return_type temp1 = w + 2.0 * k;
+        T_return_type temp2 = w - 2.0 * k;
+
+        fplus = log_sum_exp(log(temp1) - pow(temp1, 2) / twot, fplus);
+        fminus = log_sum_exp(log(-temp2) - pow(temp2, 2) / twot, fminus);
+      }
+    }
+    fplus = log_sum_exp(log(w) - pow(w, 2) / twot, fplus);
+    ans = lg1
+          + (-0.5 * LOG_TWO - LOG_SQRT_PI - 1.5 * log(q_asq)
+             + log_diff_exp(fplus, fminus));
+    // if large t is better
+  } else {
+    T_return_type fplus = NEGATIVE_INFTY;
+    T_return_type fminus = NEGATIVE_INFTY;
+    T_return_type halfq = q_asq / 2.0;
+    for (size_t k = static_cast<size_t>(kll); k >= 1; k--) {
+      T_return_type temp = k * pi();
+      T_return_type check = sin(temp * w);
+      if (check > 0)
+        fplus = log_sum_exp(log(static_cast<T_return_type>(k))
+                                - pow(temp, 2) * halfq + log(check),
+                            fplus);
+      else
+        fminus = log_sum_exp(log(static_cast<T_return_type>(k))
+                                 - pow(temp, 2) * halfq + log(-check),
+                             fminus);
+    }
+    if (fplus < fminus) {
+      ans = NEGATIVE_INFTY;
+    } else {
+      ans = lg1 + log_diff_exp(fplus, fminus) + LOG_PI;
+    }
+  }
+  return ans;
+}
+//-----------------------------------------------
+
+// d/dt DENSITY
+// calculate derivative of density with respect to t (in log, ans =
+// d/dt(log(f))=d/dt f'/f; ans*exp(ld)=f' on normal scale)
+template <typename T_y, typename T_alpha, typename T_delta, typename T_beta,
+          typename T_sv>
+return_type_t<T_y, T_alpha, T_delta, T_sv, T_beta> dtdwiener5(
+    const T_y& q, const T_alpha& a, const T_delta& vn, const T_beta& wn,
+    const T_sv& sv, const double& err) {
+  using T_return_type = return_type_t<T_y, T_alpha, T_delta, T_beta, T_sv>;
```

Comment:
Some questions here:
1) Do I understand it correctly to only have one function to call instead of all the subfunctions? I think this is not possible, since we call each single subfunction from the functions in wiener7_lpdf. Could you elaborate on your idea?
2) We also calculate the 4-parameter model from the wiener5-functions. Shall we change the ordering of the arguments, such that it corresponds to the ordering in wiener_lpdf? Then users could just change the function-call in their application from wiener_lpdf to wiener_full_lpdf and use the other function.